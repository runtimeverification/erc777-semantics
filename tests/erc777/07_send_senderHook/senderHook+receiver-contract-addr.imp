test.preamble();

//Setup:
//    - sender 1 has accepting sender hook
//    - sender 2 has rejecting sender hook
//    - receiver is a contract address with no hooks (account 10)
//Operations: send from both senders to the receiver.
//    - sender hook called in both cases, both revert.

test.registerTokensToSendHook(1, 1);
test.registerTokensToSendHook(2, 5);  //rejecting

test.makeCaller(8);
test.operatorSend(1, 10, 5, "", "");
test.operatorSend(2, 10, 5, "", "");
