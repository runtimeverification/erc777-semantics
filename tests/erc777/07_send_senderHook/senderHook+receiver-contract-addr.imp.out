Registering holder 1 with hook 1
Registering holder 2 with hook 5
Changing the caller to 8
8 is sending 5 from 1 to 10 (data:""),  (operatorData:"")
tokensToSend: Hook address = 1, Operator = 8, From = 1, To = 10, Value = 5, Data = , OperatorData = 
tokensToSend: sender balance: 10
tokensToSend: receiver balance: 100
REVERT
Balance of 1 is 10
Balance of 10 is 100
8 is sending 5 from 2 to 10 (data:""),  (operatorData:"")
tokensToSend: Hook address = 5, Operator = 8, From = 2, To = 10, Value = 5, Data = , OperatorData = 
tokensToSend: sender balance: 20
tokensToSend: receiver balance: 100
Reverting...
REVERT
Balance of 2 is 20
Balance of 10 is 100
